string name
string product
---
bool ret

